### PR TITLE
Release notes for 2018.07

### PR DESCRIPTION
**Pull request by cladmi**
### Contribution description

This is the draft PR for the 2018.07 release note.

I invite maintainers to push directly to my branch to document major changes that are not yet listed.
Developers can comment here to add their work.

As a reference, I tried to list all merged PRs in master and closed issues for this release in https://github.com/RIOT-OS/RIOT/milestone/23?closed=1.
It will be again updated with time.

### Issues/PRs references

The upcoming release

### CODE REVIEW

**Comment by miri64** [comment]
Any reason why you didn't go for the (IMHO more convenient) Etherpad way we usually went for?

**Comment by cladmi** [comment]
I did not remember it was an etherpad before but I thought about it. As I did it in a train with a limited connection it was easier to use vim than an online service which lead me to this. I can replace by a pad if you want.

**Comment by miri64** [comment]
No, let's try it like this this time :-)

File: release-notes.txt [hunk 1]
```diff
@@ -1,3 +1,149 @@
...
+    + Fix undefined doxygen groups and add a dedicated static check
+    + Extend `auto_init` documentation
+
+Core
+----
+    + core: clist_foreach() now returns break-causing node
+
+System libraries
+----------------
+    +
```
Comment (PeterKietzmann) [inline release-notes.txt]: Please add:
```
+    + puf_sram: SRAM PUF feature to seed PRNGs cortex-m platforms
+    + random: SHA-1 based PRNG
```

File: release-notes.txt [hunk 2]
```diff
@@ -1,3 +1,149 @@
...
+    + LUA: **TODO** tell the current state
+    -
+
+Boards
+------
+    + Nucleo boards use their marketing names
+    + HiFive1 (RISC-V)
+    + slstk3402a
+    + stm32mindev
+
```
Comment (aabadie) [inline release-notes.txt]: stm32f429i-disc1 and stm32l476-disco

File: release-notes.txt [hunk 3]
```diff
@@ -1,3 +1,149 @@
...
+CPU
+---
+    + E310 (RISC-V)
+
+Device Drivers
+--------------
+    +
+
```
Comment (aabadie) [inline release-notes.txt]: periph-eeprom

File: release-notes.txt [hunk 4]
```diff
@@ -1,3 +1,149 @@
...
+Networking
+----------
+    + asymcute: an asynchronous MQTT-SN client
+    + gnrc_ipv6: full duplicate address detection support according to
+      [RFC4862](https://tools.ietf.org/html/rfc4862)
+    + gnrc_ipv6: support to handle RDNSS option in neighbor discovery
+    + netif.h: a network-stack agnostic API to network interfaces
+    * sock_udp: allocate ephemeral on sock creation when port is 0
+    * **TODO* gcoap: List improvements
+    -
```
Comment (aabadie) [inline release-notes.txt]: Not sure but I think the API changes of the lorawan package was added in this cycle. Same for link_check feature

**Comment by cladmi** [comment]
@aabadie can you write it in a release-note style ?

For the milestone, I listed everything before around the 10 of July but did not have time to update it yet but I plan to do it.
And if it is not in the milestone, it requires to keep an offline list somewhere which is the same work, so I prefer updating the milestone.

File: release-notes.txt (hunk 4 above)
Comment (miri64) [reply release-notes.txt]: `git log` agrees with you ;-):

```sh
$ git log --oneline -Slorawan 2018.07-devel..2018.07-branch
cf8ce68 tests/pkg_semtech_loramac: improve test documentation
c9f6013 examples: add a lorawan OTAA/class A example
5b76fdf pkg/semtech-loramac: refactor API to make it thread-safe
```

**Comment by aabadie** [comment]
> can you write it in a release-note style ?

Why a keyboard and a computer it's simpler now ;) I added other things and directly pushed to your branch.

**Comment by gebart** [comment]
I added some of my work from the period

**Comment by cladmi** [comment]
I am finishing to write but do not know how to list the number of contributors… I may omit it for the moment.

**Comment by miri64** [comment]
> I am finishing to write but do not know how to list the number of contributors… I may omit it for the moment.

If you remove the `head -n 5` from 6. [here](https://github.com/RIOT-OS/RIOT/wiki/release-statistics) you get a list of all contributors. It might be that there are some duplicates (since people use different e-mail addresses sometimes) in there so don't just take the list at face value.

**Comment by cladmi** [comment]
> If you remove the head -n 5 from 6. here you get a list of all contributors. It might be that there are some duplicates (since people use different e-mail addresses sometimes) in there so don't just take the list at face value.

Thank you, I added a line in https://github.com/RIOT-OS/RIOT/wiki/%5Bdraft%5D-Managing-a-Release#actual-release

**Comment by miri64** [comment]
Is it on purpose that the "Fixed issues from last release" section is missing since the last release again?

**Comment by cladmi** [comment]
I copy pasted the last version so I missed it.

**Comment by miri64** [comment]
> I copy pasted the last version so I missed it.

For the 2017.10 release there was also a "Fixed issues from the last 3 releases" section. Maybe we can do that again and find out which issues were fixed for the since the 2018.01 release.

**Comment by cladmi** [comment]
The last release Milestone is not even complete or barely up to date it has 128 issues and PRs https://github.com/RIOT-OS/RIOT/milestone/22 when the release notes say 390 + 54. I do not have the motivation and time to fix this now…

I should manage to release soon or will not be able to support it any longer. So I will list the major closed issues for this release, add the introduction text but cannot do really more.

And I starte

**Comment by miri64** [comment]
Do as your stress level sees fit ;-)

**Comment by cladmi** [comment]
Ready for review. I will rebase when you you like the notes.

File: release-notes.txt [hunk 5]
```diff
@@ -1,3 +1,289 @@
...
+
+#8112 crypto/ccm: fix input_len checking with length_encoding
+#8532 kinetis: timer_set_absolute randomly misses its target
+#8778 sys/random: random_uint32_range is only uniform if range is a power of 2
+      bug quality defect
+#9065 flashpage_write broken on iotlab-m3
+#9154 cpu/lpc2387: Fixed broken SPI driver
+#9451 Makefile.include: fully define BASELIBS before using its value
+#9579 stm32_common: fix ram length for MCUs with non-contiguous ram regions
+#9675 tests: gnrc_netif: fix group join iteration macro
```
Comment (miri64) [inline release-notes.txt]: I think a fix for a test is not worth highlighting here.

Comment (miri64) [reply release-notes.txt]: (but I won't block because of it)

Comment (cladmi) [reply release-notes.txt]: It was a broken test in the last release spec that's why I did included something about it. Maybe I could link another one ?

Comment (miri64) [reply release-notes.txt]: Ok, then leave it in.

**Comment by cladmi** [comment]
I rebased/squashed and included all release-notes contributers by github username in the commit message.

**Comment by miri64** [comment]
Please backport.

**Comment by cladmi** [comment]
Backport provided in #9769

**Pull request merged**